### PR TITLE
Fix multi-tile bcl file faking

### PR DESCRIPTION
**Pull request by jacarey**
### Description

The file faker for a multi-tile bcl was not actually faking files. In addition, the bci file for a multi-tile bcl was not being created.

This should allow `CheckIlluminaDirectory` with `--FAKE_FILES true` to properly fake files for missing cycles.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [x] Added or modified tests to cover changes and any new functionality
- [x] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/illumina/parser/fakers/MultiTileBclFileFaker.java [hunk 1]
```diff
@@ -29,4 +38,25 @@ protected boolean addLeadingZeros() {
     protected int bufferSize() {
         return ((size - Integer.SIZE) * tiles.size()) + Integer.SIZE;
     }
-}
+
+    public void fakeMultiTileBclFile(final File bcl, final TileIndex tileIndex) throws IOException {
+        this.tiles = tileIndex.getTiles();
+        final OutputStream outputStream;
+        if (BclReader.isGzipped(bcl)) outputStream = new GZIPOutputStream(new FileOutputStream(bcl));
```
Comment (gbggrant) [inline src/main/java/picard/illumina/parser/fakers/MultiTileBclFileFaker.java:45]: I think the picard standard is to not do single line if statements:
```suggestion
        if (BclReader.isGzipped(bcl)) {
            outputStream = new GZIPOutputStream(new FileOutputStream(bcl));
        }...

```

**Review by gbggrant — APPROVED** [review]
Looks good.   (once you've checked in that missing test file).

**Comment by gbggrant** [comment]
Will this address: https://github.com/broadinstitute/picard/issues/1638 ?

**Comment by jacarey** [comment]
> Will this address: #1638 ?

Yes

**Comment by gbggrant** [comment]
Hi @jacarey looks like a compilation error?

**Comment by jacarey** [comment]
> Hi @jacarey looks like a compilation error?

:-1: I'll take a look

**Comment by jacarey** [comment]
Sorry @gbggrant not sure why this is working locally. Didn't mean to resubmit. I'm double checking now.

**Pull request merged**